src/System.Device.Gpio/System/Device/Pwm/Channels/PwmChannel.Linux.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Device.Pwm
+{
+    public partial class PwmChannel
+    {
+        /// <summary>
+        /// Creates a PWM channel running on Unix.
+        /// </summary>
+        /// <param name="pwmChip">The PWM chip number.</param>
+        /// <param name="pwmChannel">The PWM channel number.</param>
+        /// <param name="frequency">The frequency in hertz.</param>
+        /// <param name="dutyCyclePercentage">The duty cycle percentage represented as a value between 0.0 and 1.0.</param>
+        /// <param name="polarity">The polarity the channel should start in.</param>
+        /// <returns>A PWM channel running on Unix.</returns>
+        public static PwmChannel Create(
+            int pwmChip,
+            int pwmChannel,
+            double frequency,
+            double dutyCyclePercentage = 0.5,
+            PwmPolarity polarity = PwmPolarity.ActiveHigh) =>
```

Comment:
what's polarity? is it just remapping duty cycle into (1-duty_cycle)?

Reply:
Hopefully the attached image can explain.

Active High is the normal (majority use-case) where it is idled with a Low value and the pulse has a High value.

**Active High (DC = 25%)** and **Active Low (DC = 75%)** are essentially the same, but the High pulse comes at different time in the signal.

The idle state is where polarity becomes more helpful as there could be different circuit configurations.  For example, one could have a RGB LED tied to Vcc.  If the channel was stopped (or set to Active High DC = 0%), the LED would be on.  Same case for type of transistors used to drive motors and such.

![PwmPolarityExample](https://user-images.githubusercontent.com/2956825/60768182-f3a5af00-a08f-11e9-9648-c48a3cf64ca7.png)

Reply:
how is that different from using `1-DutyCycle` except everything being shifted by 1 cycle on the chart?

Reply:
> how is that different from using 1-DutyCycle except everything being shifted by 1 cycle on the chart?

More details https://github.com/dotnet/iot/pull/576#discussion_r301008653